pkg/api/v1/types.go

```diff
@@ -2030,6 +2030,9 @@ type SecurityContext struct {
 
 	// RunAsUser is the UID to run the entrypoint of the container process.
 	RunAsUser *int64 `json:"runAsUser,omitempty" description:"the user id that runs the first process in the container; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`
+
+	// Run the container using the host ipc mode
+	UseHostIpc *bool `json:"useHostIpc,omitempty" description:"run the container using the host ipc mode; see http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context"`
```

Comment:
I'm not sure it makes sense for this to be per-container. In general, I envision pods sharing all namespaces other than the mount namespace. This is why hostNetwork isn't in SecurityContext. cc @vishh 

Perhaps we need a pod-level SecurityContext? I'm not keen on having several fields of the form hostFoo directly in the podSpec.

Nit: Name of the field should be hostIPC -- IPC is an acronym.